src/Adapter/OrderReturn/QueryHandler/GetOrderDetailCustomizationsHandler.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\OrderReturn\QueryHandler;
+
+use Order;
+use PrestaShop\PrestaShop\Adapter\Entity\OrderDetail;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\Query\GetOrderDetailCustomizations;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\QueryHandler\GetOrderDetailCustomizationsHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\QueryResult\OrderDetailCustomization;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\QueryResult\OrderDetailCustomizations;
+use Product;
+
+class GetOrderDetailCustomizationsHandler implements GetOrderDetailCustomizationsHandlerInterface
+{
+    /**
+     * @var int
+     */
+    private $contextLangId;
+
+    /**
+     * GetOrderDetailCustomizationsHandler constructor.
+     *
+     * @param int $contextLangId
+     */
+    public function __construct(
+        int $contextLangId
+    ) {
+        $this->contextLangId = $contextLangId;
+    }
+
+    /**
+     * @param GetOrderDetailCustomizations $query
+     *
+     * @return OrderDetailCustomizations|null
+     *
+     * @throws \PrestaShopDatabaseException
+     * @throws \PrestaShopException
+     */
+    public function handle(GetOrderDetailCustomizations $query): ?OrderDetailCustomizations
+    {
+        $orderDetail = new OrderDetail($query->getOrderDetailId()->getValue());
```

Comment:
Please create repositories for this, with their interfaces in the core, to avoid code duplication. And to ease refactoring later. 

Interfaces should return model objects (inside the core namespace). And their implementation should be placed inside the Adapter namespace.


`new Order``Product::getAllCustomizedDatas`